File: gatling-performance-suite/src/test/scala/com/example/performance/simulations/LoadSimulation.scala
package com.example.performance.simulations

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import scala.concurrent.duration._
import com.example.performance.scenarios.UserJourney
import com.example.performance.config.TestConfig

class LoadSimulation extends Simulation {

  val httpProtocol = http
    .baseUrl(TestConfig.baseUrl)
    .acceptHeader("text/html")
    .userAgentHeader("Mozilla/5.0")

  setUp(
    UserJourney.scenarioDef.inject(
      rampUsers(50).during(30.seconds),
      constantUsersPerSec(10).during(1.minute)
    )
  ).protocols(httpProtocol)
   .assertions(
     global.responseTime.percentile3.lt(2000),
     global.successfulRequests.percent.gt(95)
   )
}

File: src/test/java/com/example/performance/simulations/LoadSimulation.java
package com.example.performance.simulations;

import static java.time.Duration.ofMinutes;
import static java.time.Duration.ofSeconds;

import com.example.performance.config.TestConfig;
import com.example.performance.scenarios.UserJourneyScenario;

import static io.gatling.javaapi.core.CoreDsl.constantUsersPerSec;
import static io.gatling.javaapi.core.CoreDsl.global;
import static io.gatling.javaapi.core.CoreDsl.rampUsers;
import io.gatling.javaapi.core.Simulation;
import static io.gatling.javaapi.http.HttpDsl.http;
import io.gatling.javaapi.http.HttpProtocolBuilder;

public class LoadSimulation extends Simulation {

    HttpProtocolBuilder httpProtocol = http
            .baseUrl(TestConfig.BASE_URL)
            .acceptHeader("text/html")
            .userAgentHeader("Mozilla/5.0");

    {
        setUp(
                UserJourneyScenario.scenarioDef.injectOpen(
                        rampUsers(50).during(ofSeconds(30)),
                        constantUsersPerSec(10).during(ofMinutes(1))
                )
        ).protocols(httpProtocol)
                .assertions(
                        global().responseTime().percentile3().lt(Integer.valueOf(2000)),
                        global().successfulRequests().percent().gt(95.0)
                );
    }
}